datafusion/optimizer/src/optimize_projections/mod.rs

```diff
@@ -819,6 +843,45 @@ pub fn is_projection_unnecessary(
     ))
 }
 
+fn plan_contains_non_cte_subquery(plan: &LogicalPlan, cte_name: &str) -> bool {
```

Comment:
I think this name is confusing; it still rejects subqueries that aren't CTEs I believe?

Is a name like `plan_contains_other_subqueries()` more accurate or am I misunderstanding the purpose?